Assignment2/Baseballhw.sql

```diff
@@ -0,0 +1,119 @@
+-- Show all the available databases
+SHOW DATABASES;
+
+-- Using the baseball database
+USE baseball;
+
+-- Show all the tables in the baseball database
+SHOW TABLES;
+
+-- Show game table
+
+SELECT *
+     FROM game
+     LIMIT 0, 20;
+
+-- Show batter_counts table
+
+SELECT *
+     FROM batter_counts
+     LIMIT 0,20;
+
+-- Adding index to game table
+ALTER TABLE game ADD INDEX `Gameid_date_ind`(`game_id`);
+
+-- Adding index to batter_counts table
+ALTER TABLE batter_counts ADD INDEX `Gameid_date_ind`(`game_id`);
+
+-- Creating a table to store batter stats
+DROP TABLE IF EXISTS s_batter_stats
+CREATE TABLE s_batter_stats
+    (INDEX Gameid_ind(game_id), INDEX Localdate(local_date), INDEX year_ind(Year))
+             SELECT baseball.game.game_id as game_id,
+                    DATE(baseball.game.local_date) as local_date,
+                    EXTRACT(YEAR FROM baseball.game.local_date) AS Year ,
+                    batter,
+                    atBat,
+                    Hit
+                    FROM batter_counts
+             JOIN baseball.game
+             ON baseball.game.game_id=batter_counts.game_id;
+
+-- Viewing the generated table
+
+SELECT *
+     FROM s_batter_stats
+     LIMIT 0,20;
+
+-- Calculating the historic average of single player
+
+SELECT batter as Player ,
+       ROUND(SUM(Hit)/SUM(atbat), 3)
+           AS Historic_BattingAverage from s_batter_stats WHERE batter=110029  ;
+
+-- Calculating historic average of each player and storing in a table
+
+DROP TABLE IF EXISTS f_historic_battingavg
+CREATE TABLE f_historic_battingavg AS
+              SELECT batter as Player ,
+                      (SELECT CASE
+                              WHEN SUM(atBat)=0
+                              THEN NULL
```

Comment:
line 61: You can use 0 instead of NULL as we'll be using this database in the future for feature engineering it will be better if we have 0 instead of NULL in these tables.

Reply:
I used null to differentiate the cases of SUM(hit)=0 and SUM(atBat)=0, any way considered this and made changes!